### ISSUE TITLE
[Feature]: Add SSL/TLS authentication support with certs for external kafka

### ISSUE THREAD

**Issue by lashsingh** [milvus-io/milvus#27563]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

Please see here: https://github.com/milvus-io/milvus/discussions/27486

### Describe the solution you'd like.

Add the option to enable SSL/TLS authentication for external kafka along with the option to provide the certifications location.

### Describe an alternate solution.

None found to make it work with the Milvus release v2.3.0

### Anything else? (Additional Context)

It'll be great to get the timeline to add this support, so we use Milvus becuase our current setup uses Kafka as a Service and only allows authentication via certificates. Thanks.

**Comment by xiaofan-luan** [comment]
@jaime0815 
did we already support this feature?

**Comment by prrs** [comment]
Referring these:
1. https://github.com/apache/kafka/pull/10007/files, and
2. https://github.com/milvus-io/milvus/commit/2009c3c7830e645eecb35579da9a5b910f1e172b

It looks to me if we are presenting the client certificate it's mTLS. TLS wouldn't require client to present it's certificate. 

If this is true, this feature should be mark as complete as current milvus(even 2.3.x) support mTLS. 

@xiaofan-luan @jaime0815 @yhmo

**Comment by xiaofan-luan** [comment]
> Referring these:
> 
> 1. https://github.com/apache/kafka/pull/10007/files, and
> 2. [2009c3c](https://github.com/milvus-io/milvus/commit/2009c3c7830e645eecb35579da9a5b910f1e172b)
> 
> It looks to me if we are presenting the client certificate it's mTLS. TLS wouldn't require client to present it's certificate.
> 
> If this is true, this feature should be mark as complete as current milvus(even 2.3.x) support mTLS.
> 
> @xiaofan-luan @jaime0815 @yhmo

Yes, that is what we thoughts. mTLS means user specify certification and we already support it
